### PR TITLE
Update key words

### PR DESCRIPTION
**Pull request by salomonl**
- Update options removed in d46b340f4766ffb344f926d28a86a15646bd6fc5.
- Add many keywords (VNS, feasibility, direction type, ...).
- Add a computational performance example in documentation.

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -100,6 +203,7 @@ mutable struct NomadOptions
 end
 
 function check_options(options::NomadOptions)
+    (0 < options.cache_size_max) ? nothing : error("NOMAD.jl error: max_cache_size must be strictly positive")
```
Comment (amontoison) [inline src/core.jl:206]:
```suggestion
    (0 < options.cache_size_max) || error("NOMAD.jl error: max_cache_size must be strictly positive")
```

File: src/core.jl [hunk 2]
```diff
@@ -109,8 +213,23 @@ function check_options(options::NomadOptions)
         end
     end
     (options.max_bb_eval > 0) ? nothing : error("NOMAD.jl error: wrong parameters, max_bb_eval must be strictly positive")
+    (options.eval_queue_sort ∈ EvalSortTypes) ? nothing : error("NOMAD.jl error: wrong parameters, eval_queue_sort must belong to EvalSortTypes, i.e. $(EvalSortTypes)")
```
Comment (amontoison) [inline src/core.jl:216]:
```suggestion
    (options.eval_queue_sort ∈ EvalSortTypes) || error("NOMAD.jl error: wrong parameters, eval_queue_sort must belong to EvalSortTypes, i.e. $(EvalSortTypes)")
```

File: docs/src/tutorial.md [hunk 3]
```diff
@@ -137,3 +137,69 @@ println("And inside bound constraints: ", all(-10.0 .<= result.x_best_feas .<= 1
 ```
 
 The reader can take a look at the [`test`](https://github.com/bbopt/NOMAD.jl/tree/master/test) folder for more complex examples.
+
+## Trade-offs for computational time performance
+
+The default parameters of `NOMAD.jl` closely follow the default parameters of the `NOMAD` software. More importantly, `NOMAD` tries
+to find the best solution possible according to the maximum budget of evaluations provided by the user.
```
Comment (amontoison) [inline docs/src/tutorial.md:144]:
```suggestion
to find the best solution according to the maximum budget of evaluations provided by the user.
```

File: docs/src/tutorial.md [hunk 4]
```diff
@@ -137,3 +137,69 @@ println("And inside bound constraints: ", all(-10.0 .<= result.x_best_feas .<= 1
...
+
+However, it can happen the user has a cheap blackbox in terms of computational time and needs a solution in a "short" amount of time.
```
Comment (amontoison) [inline docs/src/tutorial.md:146]:
```suggestion
However, it happens that the user has a cheap blackbox in terms of computational time and needs a solution in a "short" amount of time.
```

File: src/core.jl [hunk 5]
```diff
@@ -292,6 +415,89 @@ Maximum of calls to eval_bb allowed. Must be positive.
 
 `20000` by default.
 
+-> `sgtelib_model_max_eval::Int`:
+
+Maximum of calls to surrogate models for each optimization of
```
Comment (amontoison) [inline src/core.jl:420]:
```suggestion
Maximum number of calls to surrogate models for each optimization of
```

File: src/core.jl [hunk 6]
```diff
@@ -109,8 +213,23 @@ function check_options(options::NomadOptions)
         end
     end
     (options.max_bb_eval > 0) ? nothing : error("NOMAD.jl error: wrong parameters, max_bb_eval must be strictly positive")
+    (options.eval_queue_sort ∈ EvalSortTypes) ? nothing : error("NOMAD.jl error: wrong parameters, eval_queue_sort must belong to EvalSortTypes, i.e. $(EvalSortTypes)")
+    (options.sgtelib_model_max_eval > 0) ? nothing : error("NOMAD.jl error: wrong parameters, sgtelib_model_max_eval must be strictly positive")
```
Comment (amontoison) [inline src/core.jl:217]: You can replace all `? nothing :` by a `||`.

Comment (salomonl) [reply src/core.jl]: Done

File: docs/src/tutorial.md (hunk 4 above)
Comment (salomonl) [reply docs/src/tutorial.md]: Done

**Review by amontoison — APPROVED** [review]
Can you prepare a PR for a new release @salomonl ?

**Comment by salomonl** [comment]
Which means adding tag and upgrading dependency packages ?

**Comment by amontoison** [comment]
> Which means adding tag and upgrading dependency packages ?

You just need to open a PR that modify the version of NOMAD.jl in the Project.toml.
TagBot will add a new tag and a new release for you.

**Comment by amontoison** [comment]
Thanks @salomonl !
I don't know how we will update the keywords without your help in the future :(

**Pull request merged**